src/giskardpy/tree_manager.py

```diff
@@ -0,0 +1,140 @@
+
+        def add_child(self, manager_node):
+            if isinstance(self.node, PluginBehavior):
+                self.enabled_children.add(manager_node)
+                self.node.add_plugin(manager_node.node)
+            else:
+                if manager_node.position < 0:
+                    manager_node.position = 0
+                    if self.enabled_children:
+                        manager_node.position = max(manager_node.position, self.enabled_children[-1].position + 1)
+                    if self.disabled_children:
+                        manager_node.position = max(manager_node.position, self.disabled_children[-1].position + 1)
+                    idx = manager_node.position
+                else:
+                    idx = self.disabled_children.bisect_left(manager_node)
+                    for c in list(self.disabled_children.islice(start=idx)):
+                        c.position += 1
+                    idx = self.enabled_children.bisect_left(manager_node)
+                    for c in list(self.enabled_children.islice(start=idx)):
+                        c.position += 1
+                self.node.insert_child(manager_node.node, idx)
+                self.enabled_children.add(manager_node)
+
+        def remove_child(self, manager_node):
+            if isinstance(self.node, PluginBehavior):
+                if manager_node in self.enabled_children:
+                    self.enabled_children.remove(manager_node)
+                    self.node.remove_plugin(manager_node.node.name)
+                elif manager_node in self.disabled_children:
+                    self.disabled_children.remove(manager_node)
+                else:
+                    raise RuntimeError('could not remove node from parent. this probably means that the tree is inconsistent')
+            else:
+                if manager_node in self.enabled_children:
+                    self.enabled_children.remove(manager_node)
+                    self.node.remove_child(manager_node.node)
+                elif(manager_node in self.disabled_children):
+                    self.disabled_children.remove(manager_node)
+                else:
+                    raise RuntimeError('could not remove node. this probably means that the tree is inconsistent')
+                idx = self.disabled_children.bisect_right(manager_node)
+                for c in list(self.disabled_children.islice(start=idx)):
+                    c.position -= 1
+                idx = self.enabled_children.bisect_right(manager_node)
+                for c in list(self.enabled_children.islice(start=idx)):
+                    c.position -= 1
+
+    def __init__(self, tree):
+        self.tree = tree
+        self.tree_nodes = {}
+        self.init_map(tree.root, None, 0)
+
+    def init_map(self, node, parent, idx):
+        manager_node = TreeManager.ManagerNode(node=node, parent=parent, position=idx)
+        if parent != None:
```

Comment:
when comparing to a singleton like `None`, you should use `is` -> `if parent is not None`

Reply:
there are a few more cases where you compare with `!=`